### PR TITLE
Fix GridIn::read_msh

### LINKED ISSUE #8276: Read mesh issue introduced by gmsh-4.1 fix
@luca-heltai @masterleinad Houston, we have a problem.

The GMSH-4.1 compatibility changes introduced by d4815f3c75bb346a278f677e4c9c251171f58088 introduce 12 failing tests [1], all related to `dealii::GridIn<dim, spacedim>::read_msh(std::istream&)`. This is a regression we have to fix, and we have to make a point release to 9.1.1 in order to address this.

  [1] https://cdash.kyomu.43-1.org/viewTest.php?onlydelta&buildid=10013

### PR DESCRIPTION
**Pull request by masterleinad**
Fixes #8276 for me testing with `gmsh-4.0.2`.

### CODE REVIEW

File: source/grid/grid_in.cc [hunk 1]
```diff
@@ -1780,49 +1780,35 @@ GridIn<dim, spacedim>::read_msh(std::istream &in)
                   >> cell_type;  // ELM-TYPE
               }
 
-            switch (gmsh_file_format)
+            if (gmsh_file_format < 20)
               {
-                case 10:
```
Comment (tamiko) [inline source/grid/grid_in.cc]: In the original code, under what condition was it possible that `gmsh_file_format` was less than 10?

Comment (tamiko) [reply source/grid/grid_in.cc]: Because looking at lines 1422 - 1428 suggests that this is not possible, or am I blind? :-D

Comment (masterleinad) [reply source/grid/grid_in.cc]: The issue was that `gmsh_file_format` was 22. Previously, "2.2" was reduced to "2" but now we multiply by 10 and store that as `gmsh_file_format`.

Comment (tamiko) [reply source/grid/grid_in.cc]: Ah, what a fluke :-D

**Pull request merged**